cusub_perception/detection_tree/include/detection_tree/detection_tree_node.hpp

```diff
@@ -0,0 +1,93 @@
+#ifndef DETECTION_TREE_NODE_SRC_DETECTION_TREE_NODE_H_
+#define DETECTION_TREE_NODE_SRC_DETECTION_TREE_NODE_H_
+
+// Logging Formats
+#include <string>
+#define DETECTION_TREE_NAME             std::string("[\033[92mDetection Tree\033[0m] ")
+#define DETECTION_TREE_WARN_START       std::string("\033[93m[WARN] ")
+#define DETECTION_TREE_COLOR_START      std::string("\033[95m")
+#define DETECTION_TREE_END              std::string("\033[0m")
+
+# define DOBJECT_PROBABILITY_ZERO       0.0
+# define DOBJECT_NOT_FOUND              -1
+
+#include <pluginlib/class_list_macros.h>
+#include "detection_tree/detection_tree.hpp"
+#include <nodelet/nodelet.h>
+#include <ros/ros.h>
+#include <ros/timer.h>
+#include <vector>
+#include <map>
+#include <string>
+#include <cmath>
+#include <darknet_ros_msgs/BoundingBoxes.h>
+#include <darknet_ros_msgs/BoundingBox.h>
+#include <std_msgs/Header.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/PoseArray.h>
+#include <geometry_msgs/Quaternion.h>
+#include <tf/transform_listener.h>
+#include <tf/transform_datatypes.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <opencv2/calib3d.hpp>
+#include <detection_tree/Dvector.h>
+
+// Camel case classes + functions, underscore variables + namespaces
+
+namespace det_tree_ns
+{
+    class DetectionTree : public nodelet::Nodelet
+    {
+        public:
+            virtual void onInit();
+        private:
+            std::string sub_name;
+            void darknetCallback(const darknet_ros_msgs::BoundingBoxesConstPtr bbs);
+            void cameraInfoCallback(const sensor_msgs::CameraInfo ci);
+            int transformBearingToOdom(geometry_msgs::PoseStamped& odom_cam_pose, cv::Mat& bearing_vec, std_msgs::Header& image_header);
+            int createDobject(detection_tree::Dvector* dv);
+            bool poseSolveDobject(Dobject* dobj, geometry_msgs::Pose& pose);
+            void dobjPubCallback(const ros::TimerEvent&);
+            void det_print(std::string str);
+            void det_print_warn(std::string str);
+
+            // Dvector Association Functions
```

Comment:
From reading it, it seems like the basic idea of association is to match any new dvecs to dobjs using euclidean distances as a "probability". Is this correct? I am also a little confused about the map within the map. Does each dvec have multiple unique versions with different scores? I thought each dobj has many dvecs, but this makes me think each dvec also has other dvecs.

Reply:
Yes, exactly that's the preliminary way. Yeah lol the map within a map was a data structure hack so I would just be able to loop everything. Yeah each object (ie the left start_gate_pole or right start gate pole or Jiangshi) is a dobject. When darknet produces a new bounding box of this object is gets thrown into the dobject.